hazelcast/src/main/java/com/hazelcast/internal/memory/MemoryAccessStrategy.java

```diff
@@ -0,0 +1,576 @@
+
+    /**
+     * Base offset of boolean[]
+     */
+    int ARRAY_BOOLEAN_BASE_OFFSET = STANDARD_MEM != null ? STANDARD_MEM.arrayBaseOffset(boolean[].class) : -1;
+
+    /**
+     * Base offset of byte[]
+     */
+    int ARRAY_BYTE_BASE_OFFSET = STANDARD_MEM != null ? STANDARD_MEM.arrayBaseOffset(byte[].class) : -1;
+
+    /**
+     * Base offset of short[]
+     */
+    int ARRAY_SHORT_BASE_OFFSET = STANDARD_MEM != null ? STANDARD_MEM.arrayBaseOffset(short[].class) : -1;
+
+    /**
+     * Base offset of char[]
+     */
+    int ARRAY_CHAR_BASE_OFFSET = STANDARD_MEM != null ? STANDARD_MEM.arrayBaseOffset(char[].class) : -1;
+
+    /**
+     * Base offset of int[]
+     */
+    int ARRAY_INT_BASE_OFFSET = STANDARD_MEM != null ? STANDARD_MEM.arrayBaseOffset(int[].class) : -1;
+
+    /**
+     * Base offset of float[]
+     */
+    int ARRAY_FLOAT_BASE_OFFSET = STANDARD_MEM != null ? STANDARD_MEM.arrayBaseOffset(float[].class) : -1;
+
+    /**
+     * Base offset of long[]
+     */
+    int ARRAY_LONG_BASE_OFFSET = STANDARD_MEM != null ? STANDARD_MEM.arrayBaseOffset(long[].class) : -1;
+
+    /**
+     * Base offset of double[]
+     */
+    int ARRAY_DOUBLE_BASE_OFFSET = STANDARD_MEM != null ? STANDARD_MEM.arrayBaseOffset(double[].class) : -1;
+
+    /**
+     * Base offset of any type of Object[]
+     */
+    int ARRAY_OBJECT_BASE_OFFSET = STANDARD_MEM != null ? STANDARD_MEM.arrayBaseOffset(Object[].class) : -1;
+
+    /////////////////////////////////////////////////////////////////////////
+
+    /**
+     * Index scale of boolean[]
+     */
+    int ARRAY_BOOLEAN_INDEX_SCALE = STANDARD_MEM != null ? STANDARD_MEM.arrayIndexScale(boolean[].class) : -1;
+
+    /**
+     * Index scale of byte[]
+     */
+    int ARRAY_BYTE_INDEX_SCALE = STANDARD_MEM != null ? STANDARD_MEM.arrayIndexScale(byte[].class) : -1;
+
+    /**
+     * Index scale of short[]
+     */
+    int ARRAY_SHORT_INDEX_SCALE = STANDARD_MEM != null ? STANDARD_MEM.arrayIndexScale(short[].class) : -1;
+
+    /**
+     * Index scale of char[]
+     */
+    int ARRAY_CHAR_INDEX_SCALE = STANDARD_MEM != null ? STANDARD_MEM.arrayIndexScale(char[].class) : -1;
+
+    /**
+     * Index scale of int[]
+     */
+    int ARRAY_INT_INDEX_SCALE = STANDARD_MEM != null ? STANDARD_MEM.arrayIndexScale(int[].class) : -1;
+
+    /**
+     * Index scale of float[]
+     */
+    int ARRAY_FLOAT_INDEX_SCALE = STANDARD_MEM != null ? STANDARD_MEM.arrayIndexScale(float[].class) : -1;
+
+    /**
+     * Index scale of long[]
+     */
+    int ARRAY_LONG_INDEX_SCALE = STANDARD_MEM != null ? STANDARD_MEM.arrayIndexScale(long[].class) : -1;
+
+    /**
+     * Index scale of double[]
+     */
+    int ARRAY_DOUBLE_INDEX_SCALE = STANDARD_MEM != null ? STANDARD_MEM.arrayIndexScale(double[].class) : -1;
+
+    /**
+     * Index scale of any type of Object[]
+     */
+    int ARRAY_OBJECT_INDEX_SCALE = STANDARD_MEM != null ? STANDARD_MEM.arrayIndexScale(Object[].class) : -1;
```

Comment:
All these constants and the `...offset` methods below don't belong to the general memory access strategy. These would have to be moved to an interface specialized for native CPU memory access. This can be done in a further PR.